fink_utils/sso/utils.py

```diff
@@ -437,6 +437,38 @@ def compute_light_travel_correction(jd, d_obs):
     return jd_lt
 
 
+def estimate_axes_ratio(residuals, R):
+    """Estimate the axes ratio of a SSO from the residuals of the sHG1G2 model
+    and its oblateness R.
+
+    Parameters
+    ----------
+    residuals: np.array
+        Residuals (observed - model) of the SSO with sHG1G2 model
+    R: float
+        Oblateness parameter of the sHG1G2 model
+
+    Returns
+    -------
+    a_b, a_c: float
+        a/b and a/c axes ratio
+    """
+
+    # Estimate the amplitude of the lightcurve from residuals
+    # Taken at 2 sigma
+    amplitude = np.std(residuals) * 2.0
+
+    # Estimate the a/b ratio
+    a_b = 10 ** (0.4 * (amplitude * 2))
+
+    # Estimate the a/c ratio (and force c<b)
+    a_c = (a_b + 1) / (2 * R)
+    if a_c < a_b:
+        a_c = a_b
```

Comment:
Just to make sure: if a/c < a/b, you want both to be equal (and not swap the two values)?

Reply:
Yes: a/c is provided by the long-term behavior (kind similar to R in sHG1G2), while a/b is linked with the short term variability (rotation). But we expect a>b>c (principal rotator of a tri-axial ellipsoid).
As these are initial conditions, we set a/c = a/b if there is an issue with the "crude" estimate

Reply:
Perfect :-)